react/components/DatePicker/index.js

```diff
@@ -0,0 +1,151 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import ReactDatePicker, { registerLocale } from 'react-datepicker'
+import * as locales from 'date-fns/locale/index.js'
+import Input from '../Input'
+import IconCalendar from '../icon/Calendar'
+
+import './react-datepicker.global.css'
+
+class DatePicker extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      active: false,
+    }
+
+    this.handleLocaleChange(props.locale)
+  }
+
+  handleChange = event => {
+    this.props.onChange && this.props.onChange(event)
+  }
+
+  handleFocus = event => {
+    this.setState({ active: true })
+    this.props.onFocus && this.props.onFocus(event)
+  }
+
+  handleBlur = event => {
+    this.setState({ active: false })
+    this.props.onBlur && this.props.onBlur(event)
+  }
+
+  handleLocaleChange = locale => {
+    registerLocale(locale, locales[locale.replace('-', '')])
```

Comment:
talvez comenta o propósito dessa funçao? parece meio esoterica

Reply:
acho que é do proprio date-fns

Reply:
vou adicionar um comentário

Reply:
@klzns na verdade essa função é do próprio react-datepicker, to tentando entender certinho o que ela faz pra deixar no comentário